### PR TITLE
refactor(connlib): remove `ConnId`

### PR DESCRIPTION
**Pull request by thomaseizinger**
There is no reason we should _ever_ use a `ClientId` in place of a `GatewayId` or `ResourceId` and vice versa. Thus, the concept of `ConnId` will just lead to bugs.

We can parameterize everything with a `TId` type and infer, what the correct ID is based on the `RoleState` trait.

### CODE REVIEW

**Comment by thomaseizinger** [comment]
Draft because it is stacked on top of https://github.com/firezone/firezone/pull/2321.

**Comment by jamilbk** [comment]
Merged main to fix CI

**Comment by thomaseizinger** [comment]
Friendly ping @conectado :)

File: rust/connlib/tunnel/src/client.rs [hunk 1]
```diff
@@ -93,7 +93,7 @@ where
     // FIXME: this cleanup connection is wrong!
     pub fn cleanup_connection(&self, id: ResourceId) {
         self.role_state.lock().on_connection_failed(id);
-        self.peer_connections.lock().remove(&id.into());
+        // self.peer_connections.lock().remove(&id.into());
```
Comment (conectado) [inline rust/connlib/tunnel/src/client.rs:140]: shouldn't we still cleanup the `peer_connections` state? or are we sure this is always called before it's inserted?

**Review by conectado — APPROVED** [review]
LGTM! left a non-blocking comment

File: rust/connlib/tunnel/src/client.rs (hunk 1 above)
Comment (thomaseizinger) [reply rust/connlib/tunnel/src/client.rs:140]: I don't know to be honest. I think the connection cleanup isn't very consistent at the moment.

I am working on a few more refactorings which should make it a lot easier to do this consistently :)

I'd suggest deferring until then!

**Pull request merged**